synapse/storage/databases/main/receipts.py

```diff
@@ -274,6 +274,49 @@ def f(txn):
         }
         return results
 
+    @cached(num_args=2,)
+    async def get_linearized_receipts_for_all_rooms(
+        self, to_key: int, from_key: Optional[int] = None
+    ):
```

Comment:
It would be ideal to add a docstring and a return type to this (which I think is `Dict[str, JsonDict]`).